src/saltext/vmware/states/vmc_org_user.py

```diff
@@ -0,0 +1,330 @@
+                        }
+                    ]
+            where 'name' indicates the name of the organization role.
+
+    skip_notify
+        (Optional) Indicates if the notification have to be skipped.
+
+    custom_roles
+        (Optional) List of custom role assigned to a user.
+        It can be specified in the below format
+
+        .. code::
+
+            "custom_roles": [
+                        {
+                            "name": "role_name"
+                        }
+                    ]
+            where 'name' indicates the name of the custom role.
+
+    service_roles
+        (Optional) List of service roles to attach to a user.
+        Below fields defines the properties of service roles.
+
+        'serviceDefinitionLink': (String) (Optional)
+            The link to the service definition.
+
+        'serviceRoles': list
+            It can be specified in the below format
+
+            .. code::
+
+                "serviceRoles": [
+                        {
+                            "name": "role_name"
+                        }
+                    ]
+                where 'name' indicates the name of the service role.
+
+    skip_notify_registration
+        (Optional) Prevent sending mails to users that do not yet have a CSP profile.
+
+    invited_by
+        (Optional) Indicates the actual user who is inviting.
+
+    custom_groups_ids
+        (Optional) List of unique IDs of Custom Groups. Can have a maximum of 15 custom group IDs.
+
+    verify_ssl
+        (Optional) Option to enable/disable SSL verification. Enabled by default.
+        If set to False, the certificate validation is skipped.
+
+    cert
+        (Optional) Path to the SSL client certificate file to connect to VMC Cloud Console.
+        The certificate can be retrieved from browser.
+
+    For example:
+
+        .. code-block:: yaml
+
+            test_user@vmware.com:
+              vmc_org_user.invite:
+                - hostname: console-stg.cloud.vmware.com
+                - refresh_key: 7jPSGSZpCa8e5Ouks4UY5cZyOtynAhF
+                - org_id: 10e1092f-51d0-473a-80f8-137652c39fd0
+                - organization_roles:
+                    - name: org_member
+                - verify_ssl: False
+                - cert: /path/to/client/certificate
+
+    """
+
+    username = name
+    org_users_list = __salt__["vmc_org_users.list"](
+        hostname=hostname,
+        refresh_key=refresh_key,
+        org_id=org_id,
+        verify_ssl=verify_ssl,
+        cert=cert,
+    )
+
+    if "error" in org_users_list:
+        return vmc_state._create_state_response(
+            name=name,
+            comment="Failed to get users for given org : {}".format(org_users_list["error"]),
+            result=False,
+        )
+    org_user = None
+    for org_user in org_users_list.get("results"):
+        if username == org_user["user"].get("username"):
+            break
+        else:
+            org_user = None
```

Comment:
**suggestion** rather than changing the org_user every time, this could be re-written to use the for/else construct:
```
for org_user in org_users_list["results"]:
    if username == org_user["name"].get("username"):  # will the user ever be missing a username?
        break
else:
    org_user = None
```

Or it could be re-written with a generator comprehension/expression:

```
org_user = next((user for user in org_users_list["results"] if user["user"].get("username") == username), None)
```

Reply:
Updated